### PR TITLE
Support GitLab URLs

### LINKED ISSUE #1: Gitlab URLs are ignored
The current script is unable to generate Gitlab URLs. While core packages in repositories like https://gitlab.com/ros-tracing/ros2_tracing can still be scrapped, these end up without a link to the source CHANGELOG.rst file.

### PR DESCRIPTION
**Pull request by christophebedard**
Fixes #1

I was getting relative `dirpath`s, which broke the logic for extracting the `url_path`. Not sure if you were having this issue.

Signed-off-by: Christophe Bedard <bedard.christophe@gmail.com>

### CODE REVIEW

**Comment by christophebedard** [comment]
@hidmic FYI

File: walk-packages-for-changelog.py [hunk 1]
```diff
@@ -391,13 +393,19 @@ def main():
...
 
         header = ''
-        if origin_url and origin_url.startswith('https://github.com'):
+        from_github = origin_url.startswith('https://github.com')
+        from_gitlab = origin_url.startswith('https://gitlab.com')
+        if origin_url and (from_github or from_gitlab):
             if origin_url.endswith('.git'):
                 origin_url = origin_url[:-4]
             url_path = '/'
             if len(package.dirpath) != len(repo_path):
```
Comment (christophebedard) [inline walk-packages-for-changelog.py:402]: I guess I could move `os.path.abspath(...)` to here, e.g.
```suggestion
            package.dirpath = os.path.abspath(package.dirpath)
            if len(package.dirpath) != len(repo_path):
```

**Comment by clalancette** [comment]
Oops!  Apologies, I did a similar fix and just pushed it to master.  I think it should be equivalent to yours (with the exception of the dirpath thing), so I'm going to close this out.  Feel free to open another issue/PR with the dirpath thing; it hasn't been required in my setup, but maybe I'm invoking it differently than you.

**Comment by christophebedard** [comment]
> Oops! Apologies, I did a similar fix and just pushed it to master.

No worries!

> Feel free to open another issue/PR with the dirpath thing; it hasn't been required in my setup, but maybe I'm invoking it differently than you.

Yeah probably because I provided a relative path for `source_path` and not an absolute path.

**Pull request closed without merging**